### DISCUSSION TITLE
GAS and ELF support

### DISCUSSION THREAD

**Discussion by atirutw** [maziac/DeZog#133] [Ideas]
Upstream GNU Binutils (and the assembler) supports Z80, and ELF files can have a lot of useful information (e.g. where to load, entry point address, etc.), so this would be very useful especially for anything involving ELF. A personal use case of mine would also benefit from this.

**Comment by maziac** [comment]
The problem with the Z80 assemblers is that they usually don't provide enough information for banking.
Historically DeZog supports z80asm, z88dk. But only sjasmplus supports proper banking information.

I was happy that the maintainer of sjasmplus was able to add this information in a new output format (additionally to the lst format).
I doubt that it will be provided by gas.

But if you would like to try you can fork dezog.
There is also documentation available for adding an assembler: https://github.com/maziac/DeZog/blob/main/design/AddingNewAssemblers.md

**Reply by atirutw** [reply]
> The problem with the Z80 assemblers is that they usually don't provide enough information for banking.
> Historically DeZog supports z80asm, z88dk. But only sjasmplus supports proper banking information.

Why would that be a problem? It's fine if a format doesn't have banking information considering you can already load flat binaries. In this case, ELF could be treated as a file with `loadObjs` information.
